### PR TITLE
[SCB-498] Configuration Center ip address need to config default port

### PR DESCRIPTION
**Pull request by weichao666**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---
for http/https, the default ports are 80 and 443, if port is not configured, the default port needs to be used. this simplifies the configuration.

### CODE REVIEW

**Comment by yangbor** [comment]
It seems to be the wrong place to fix the problem. the http(80) and https(443) are Well Known Ports and all http servers and clients should handle them correctly. We should fix the code where this is not correctly handled instead of appending the port to the URL.

**Comment by weichao666** [comment]
modified

File: foundations/foundation-common/src/main/java/org/apache/servicecomb/foundation/common/net/NetUtils.java [hunk 1]
```diff
@@ -145,12 +145,25 @@ public static IpPort parseIpPortFromURI(String uriAddress) {
...
   }
 
+  private static IpPort parseIpPort(String scheme, String authority) {
+    if (authority == null)
+      return null;
+    int idx = authority.indexOf(':');
+    if (idx != -1)
+      return parseIpPort(authority);
+    if (scheme.equals("http"))
+      return parseIpPort(new StringBuffer(authority).append(":80").toString());
```
Comment (liubao68) [inline foundations/foundation-common/src/main/java/org/apache/servicecomb/foundation/common/net/NetUtils.java]: return new IpPort(authority, 80)

File: foundations/foundation-common/src/main/java/org/apache/servicecomb/foundation/common/net/NetUtils.java [hunk 2]
```diff
@@ -145,12 +145,25 @@ public static IpPort parseIpPortFromURI(String uriAddress) {
...
+    if (scheme.equals("https"))
+      return parseIpPort(new StringBuffer(authority).append(":443").toString());
+    return parseIpPort(authority);
```
Comment (liubao68) [inline foundations/foundation-common/src/main/java/org/apache/servicecomb/foundation/common/net/NetUtils.java]: return null;

Comment (weichao666) [reply foundations/foundation-common/src/main/java/org/apache/servicecomb/foundation/common/net/NetUtils.java]: only http and https was deal with here, if it's starts with others, e.g "rest://",shouldn't return null

File: foundations/foundation-common/src/main/java/org/apache/servicecomb/foundation/common/net/NetUtils.java (hunk 1 above)
Comment (weichao666) [reply foundations/foundation-common/src/main/java/org/apache/servicecomb/foundation/common/net/NetUtils.java]: done

File: foundations/foundation-common/src/main/java/org/apache/servicecomb/foundation/common/net/NetUtils.java (hunk 2 above)
Comment (weichao666) [reply foundations/foundation-common/src/main/java/org/apache/servicecomb/foundation/common/net/NetUtils.java]: done

File: sh.exe.stackdump [hunk 3]
```diff
@@ -0,0 +1,11 @@
+Stack trace:
```
Comment (wujimin) [inline sh.exe.stackdump]: why we need this file?

File: foundations/foundation-common/src/test/java/org/apache/servicecomb/foundation/common/net/TestNetUtils.java [hunk 4]
```diff
@@ -49,6 +49,10 @@ public void testNetutils() {
     Assert.assertEquals(NetUtils.parseIpPortFromURI(null), null);
     Assert.assertEquals(NetUtils.parseIpPortFromURI("ss"), null);
     Assert.assertEquals(NetUtils.parseIpPortFromURI("rest://127.0.0.1:8080").getHostOrIp(), "127.0.0.1");
+    Assert.assertEquals(NetUtils.parseIpPortFromURI("http://127.0.0.1:8080").getPort(), 8080);
```
Comment (wujimin) [inline foundations/foundation-common/src/test/java/org/apache/servicecomb/foundation/common/net/TestNetUtils.java]: expect value is the first parameter

Comment (weichao666) [reply foundations/foundation-common/src/test/java/org/apache/servicecomb/foundation/common/net/TestNetUtils.java]: done

**Pull request merged**